### PR TITLE
Register Expresss Checkout block only when enabled in the settings

### LINKED ISSUE #9444: Google & Apple Pay buttons are displayed in editor even when the payment method is disabled
### Describe the bug

The Google & Apple Pay buttons are displayed in editor even when the payment method is disabled in WooPayments settings.

### To Reproduce

1. In wp-admin, go to WooCommerce > Settings > Payments > WooPayments page
2. Disable Apple / Google Pay
3. Open the checkout page in the editor
4. See that the Apple & Google pay buttons are still rendered in the editor

### Actual behavior

The Apple and Google Pay buttons are visible in the editor when they are disabled in settings.

### Expected behavior

The Apple and Google Pay buttons should not be visible in the editor when they are disabled in settings.

### Potential solution
We should only call `registerExpressPaymentMethod` if it is enabled from the WooPayments settings page

### PR DESCRIPTION
**Pull request by gpressutto5**
Fixes #9444

#### Changes proposed in this Pull Request

The Google & Apple Pay buttons were displayed in editor even when the payment method is disabled in WooPayments settings because they were always registered. This PR adds a check before registering them.

#### Testing instructions

1. In wp-admin, go to WooCommerce > Settings > Payments > WooPayments page
1. Disable Apple / Google Pay
1. Open the checkout page in the editor
1. See that the Apple & Google pay buttons are still rendered in the editor when using `develop`
1. Checkout this branch and ensure they are not rendered anymore
1. Enable Apple / Google Pay and ensurethey are rendered
1. Smoke test to ensure everything still work as exptected

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9646` or branch name `fix/express-checkout-block-registering` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/express-checkout-block-registering) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: d2af7364fdb68733d0635ed0effe331c10e9bb04
- Build time: 2024-10-30 15:00:58 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/checkout/blocks/index.js [hunk 1]
```diff
@@ -157,15 +157,17 @@ if ( getUPEConfig( 'isWooPayEnabled' ) ) {
...
+		registerExpressPaymentMethod(
+			tokenizedCartPaymentRequestPaymentMethod( api )
+		);
+	} else if ( getUPEConfig( 'isExpressCheckoutElementEnabled' ) ) {
+		registerExpressPaymentMethod( expressCheckoutElementApplePay( api ) );
+		registerExpressPaymentMethod( expressCheckoutElementGooglePay( api ) );
+	} else {
+		registerExpressPaymentMethod( paymentRequestPaymentMethod( api ) );
+	}
 }
```
Comment (timur27) [inline client/checkout/blocks/index.js:171]: This is a matter of preference, but I started to find the below approach more readable after some time in our back-end PHP:

```suggestion
if (!getUPEConfig('isPaymentRequestEnabled')) {
    return;
}

if (getUPEConfig('isTokenizedCartPrbEnabled')) {
    registerExpressPaymentMethod(tokenizedCartPaymentRequestPaymentMethod(api));
    return;
}

if (getUPEConfig('isExpressCheckoutElementEnabled')) {
    registerExpressPaymentMethod(expressCheckoutElementApplePay(api));
    registerExpressPaymentMethod(expressCheckoutElementGooglePay(api));
    return;
}

registerExpressPaymentMethod(paymentRequestPaymentMethod(api));
```

feel free to ignore it if the change does not seem convincing.

**Review by timur27 — APPROVED** [review]
Clear change and tests well, I just left one minor comment, up to you to decide on whether you want to apply it or not. Thanks @gpressutto5

File: client/checkout/blocks/index.js (hunk 1 above)
Comment (gpressutto5) [reply client/checkout/blocks/index.js:171]: It is indeed more readable, but we have more code after that we want to execute regardless of PRB/ECE settings. If we want to early return we need to create a new function for this so it doesn't affect the rest of the code. But then I think it starts to get less readable, what do you think?

```suggestion
const maybeRegisterExpressCheckoutMethod = () => {
	if (!getUPEConfig('isPaymentRequestEnabled')) {
	    return;
	}
	
	if (getUPEConfig('isTokenizedCartPrbEnabled')) {
	    registerExpressPaymentMethod(tokenizedCartPaymentRequestPaymentMethod(api));
	    return;
	}
	
	if (getUPEConfig('isExpressCheckoutElementEnabled')) {
	    registerExpressPaymentMethod(expressCheckoutElementApplePay(api));
	    registerExpressPaymentMethod(expressCheckoutElementGooglePay(api));
	    return;
	}
	
	registerExpressPaymentMethod(paymentRequestPaymentMethod(api));
}
maybeRegisterExpressCheckoutMethod();
```

Comment (timur27) [reply client/checkout/blocks/index.js:171]: Yeah, agreed, I'd keep it as it is then without changing your final version 👍 🚢

**Comment by alexflorisca** [comment]
Thanks for the review ping, just got round to it but looks like I'm a bit late. Great to see this shipped though, nice work! 🎉

**Pull request merged**